src/Core/CoreKernel.php

```diff
@@ -116,6 +117,27 @@ protected function getDatabaseConfig()
             "password" => Environment::getEnv('SS_DATABASE_PASSWORD') ?: null,
         ];
 
+        // Only add SSL keys in the array if there is an actual value associated with them
+        $sslConf = [
+            'ssl_key' => 'SS_DATABASE_SSL_KEY',
+            'ssl_cert' => 'SS_DATABASE_SSL_CERT',
+            'ssl_ca' => 'SS_DATABASE_SSL_CA',
+            'ssl_cipher' => 'SS_DATABASE_SSL_CIPHER',
+        ];
+        foreach ($sslConf as $key => $envVar) {
+            $envValue = Environment::getEnv($envVar);
+            if ($envValue) {
+                $databaseConfig[$key] = $envValue;
+            }
+        }
+
+        // Having only the key or cert without the other is bad configuration.
+        if (isset($databaseConfig['ssl_key']) xor isset($databaseConfig['ssl_cert'])) {
+            user_error('Database SSL cert and key must both be defined to use SSL in the database.', E_USER_WARNING);
+            unset($databaseConfig['ssl_key']);
+            unset($databaseConfig['ssl_cert']);
+        }
```

Comment:
A warning instead of an exception because if a live environment is poorly configured this shouldn't necessarily break the entire site - but it _should_ provide a warning in the logs.

Note that the previous implementation didn't even give a warning, so this is an improvement over what would have been here if SSL support had never been erroneously removed in the first place.